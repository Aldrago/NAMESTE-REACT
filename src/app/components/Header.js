import { useState } from "react";
import { LOGO_URL } from "../utils/constant";
import { Link } from "react-router-dom";
import { useSelector } from "react-redux";

const Header = () => {
  const [authBtn, setAuthBtn] = useState("Login");
  const items = useSelector((store) => store.cart.items);

  return (
    <div className="flex justify-between bg-yellow-100">
      <div className="w-32">
        <img src={LOGO_URL} />
      </div>
      <div className="flex items-center">
        <ul className="flex p-4 m-4">
          <li className="px-4 text-xl"><Link to="/">Home</Link></li>
          <li className="px-4 text-xl"><Link to="/about">About us</Link></li>
          <li className="px-4 text-xl"><Link to="/contact">Contact us</Link></li>
          <li className="px-4 text-xl"><Link to="/cart">Cart -({items.length} items)</Link></li>
          <li className="px-4 py-1 text-lg bg-gray-100 rounded-lg">
            <button className="login-btn"
              onClick={() => {
                authBtn === "Login"
                  ? setAuthBtn("Logout")
                  : setAuthBtn("Login");
              }}
            >
              {authBtn}
            </button>
          </li>
        </ul>
      </div>
    </div>
  );
};
export default Header;
