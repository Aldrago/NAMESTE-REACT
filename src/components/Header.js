import { useState } from "react";
import { LOGO_URL } from "../utils/constant";
import { Link } from "react-router-dom";

const Header = () => {
  const [authBtn, setAuthBtn] = useState("login");

  return (
    <div className="flex justify-between bg-yellow-100">
      <div className="w-32">
        <img src={LOGO_URL} />
      </div>
      <div className="flex items-center">
        <ul className="flex p-4 m-4">
          <li className="px-4 text-xl"><Link to="/">Home</Link></li>
          <li className="px-4 text-xl"><Link to="/about">About us</Link></li>
          <li className="px-4 text-xl"><Link to="/contact">Contact us</Link></li>
          <li className="px-4 text-xl"><Link>Cart</Link></li>
          <li className="px-4 py-1 text-lg bg-gray-100 rounded-lg">
            <button className="login-btn"
              onClick={() => {
                authBtn === "login"
                  ? setAuthBtn("logout")
                  : setAuthBtn("login");
              }}
            >
              {authBtn}
            </button>
          </li>
        </ul>
      </div>
    </div>
  );
};
export default Header;
