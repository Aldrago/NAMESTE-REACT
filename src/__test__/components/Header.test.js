import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import appStore from "../../app/utils/appStore";
import Header from "../../app/components/Header";

describe("Header Component test cases", () => {
  test("Should match snapshort", () => {
    const { container } = render(
      <Provider store={appStore}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  test("Should contain logout on login click", () => {
    render(
      <Provider store={appStore}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>
    );

    const loginBtn = screen.getByRole("button", { name: "Login" });
    fireEvent.click(loginBtn);
    const logoutBtn = screen.getByRole("button", { name: "Logout" });

    expect(logoutBtn).toBeInTheDocument();
  });
});
