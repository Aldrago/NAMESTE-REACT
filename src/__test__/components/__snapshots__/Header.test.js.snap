// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component test cases Should match snapshort 1`] = `
<div>
  <div
    class="flex justify-between bg-yellow-100"
  >
    <div
      class="w-32"
    >
      <img
        src="https://static.vecteezy.com/system/resources/previews/007/890/977/original/burger-logo-for-fast-food-business-free-vector.jpg"
      />
    </div>
    <div
      class="flex items-center"
    >
      <ul
        class="flex p-4 m-4"
      >
        <li
          class="px-4 text-xl"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="px-4 text-xl"
        >
          <a
            href="/about"
          >
            About us
          </a>
        </li>
        <li
          class="px-4 text-xl"
        >
          <a
            href="/contact"
          >
            Contact us
          </a>
        </li>
        <li
          class="px-4 text-xl"
        >
          <a
            href="/cart"
          >
            Cart -(
            0
             items)
          </a>
        </li>
        <li
          class="px-4 py-1 text-lg bg-gray-100 rounded-lg"
        >
          <button
            class="login-btn"
          >
            Login
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
